synapse/lib/autodoc.py

```diff
@@ -54,6 +54,8 @@
                          'description': 'The name of the object.'},
                 'desc': {'type': 'string',
                          'description': 'The docstring of the object.'},
+                'deprecated': {'type': 'boolean',
```

Comment:
Can we add this to the rendered output we generate in the autodocs?

Reply:
I can yea. You're thinking just like a Strikethrough on the storm locals?

Reply:
Rendering a strikethrough may require additional sphinx extensions.
We may be able to just get by with making sure the desc for the deprecated functions and libs contain the word `deprecated` somehow to make sure its clear that they are deprecated, esp since we may have alternatives to suggest?

Reply:
How about I add a small "Foo is deprecated" blurb that we can wordsmith to the autodoc generation code? The description (sometimes) contains the alternative.

Reply:
Lets give it a spin and see how it looks :)

Reply:
Well I pushed it so tell me what you think

Reply:
:eyes: